ouroboros-network/src/Ouroboros/Network/NodeToClient.hs

```diff
@@ -673,3 +683,35 @@ networkErrorPolicies = ErrorPolicies
     shortDelay = 20 -- seconds
 
 type LocalConnectionId = ConnectionId LocalAddress
+
+--
+-- Null Protocol Peers
+--
+
+chainSyncPeerNull
+    :: forall (header :: Type) (tip :: Type) m a. MonadTimer m
+    => Peer (ChainSync.ChainSync header tip)
+            AsClient ChainSync.StIdle m a
+chainSyncPeerNull =
+    ChainSync.chainSyncClientPeer
+      (ChainSync.ChainSyncClient untilTheCowsComeHome )
+
+localStateQueryPeerNull
+    :: forall (block :: Type) (query :: Type -> Type) m a. MonadTimer m
+    => Peer (LocalStateQuery.LocalStateQuery block query)
+            AsClient LocalStateQuery.StIdle m a
+localStateQueryPeerNull =
+    LocalStateQuery.localStateQueryClientPeer
+      (LocalStateQuery.LocalStateQueryClient untilTheCowsComeHome)
+
+localTxSubmissionPeerNull
+    :: forall (tx :: Type) (reject :: Type) m a. MonadTimer m
+    => Peer (LocalTxSubmission.LocalTxSubmission tx reject)
+            AsClient LocalTxSubmission.StIdle m a
+localTxSubmissionPeerNull =
+    LocalTxSubmission.localTxSubmissionClientPeer
+      (LocalTxSubmission.LocalTxSubmissionClient untilTheCowsComeHome)
+
+-- ;)
+untilTheCowsComeHome :: MonadTimer m => m a
+untilTheCowsComeHome = forever $ threadDelay 43200 {- day in seconds -}
```

Comment:
43200 is only 12 hours in seconds.
It may make sense to add a new function `threadDelayForever` to `MonadDelay`
and maybe use a different Implementation then `forever $ threadDelay 43200` (maybe `atomically retry`). 
For debugging etc it may be useful to detect that something is blocked forever (`threadDelayForever`).

Reply:
Isn't waking up once every 12 hours enough? The cost is negligible considering `GHz` CPU frequency.

> It may make sense to add a new function threadDelayForever to MonadDelay

I think we want to keep `io-sim-classes` as small as possible, preferably only exposing api's that are in `base` / `async` / `stm`.

> For debugging etc it may be useful to detect that something is blocked forever (threadDelayForever).

This would be a programmer error. For this reason I prefer to use `Void`, e.g. `loop  = forever someIO :: IO Void` to make it clear at the type level that `loop` never returns.